data-prepper-plugin-framework/src/main/java/org/opensearch/dataprepper/plugin/DefaultPluginFactory.java

```diff
@@ -115,13 +116,16 @@ private <T> ComponentPluginArgumentsContext getConstructionContext(final PluginS
         final PluginConfigObservable pluginConfigObservable = pluginConfigurationObservableFactory
                 .createDefaultPluginConfigObservable(pluginConfigurationConverter, pluginConfigurationType, pluginSetting);
 
+        Class[] markersToScanForDI = pluginAnnotation.packagesToScanForDI();
```

Comment:
Please verify this behavior in the unit tests for this class.

Reply:
Added additional unit test to verify this change